examples/blog/routes/[slug].tsx

```diff
@@ -0,0 +1,81 @@
+/** @jsx h */
+import { h } from "preact";
+import { tw } from "@twind";
+import { Handlers, PageProps } from "$fresh/server.ts";
+import Layout from "components/Layout.tsx";
+import { Marked } from "https://deno.land/x/markdown@v2.0.0/mod.ts";
+import { PostDetail } from "types/blog.ts";
+import { fetchRepoMetadata } from "utils/github.ts";
+import GithubRepo from "components/GithubRepo.tsx";
+import Tag from "../components/Tag.tsx";
+import BlogAuthor from "../components/BlogAuthor.tsx";
+
+export const handler: Handlers = {
+  async GET(req, ctx) {
+    try {
+      const BLOG_DIR = "blog";
+      const slug = ctx.params.slug;
+
+      const postPath = `${BLOG_DIR}/${slug}.md`;
+      // check file exists
+      await Deno.stat(postPath);
+
+      const decoder = new TextDecoder("utf-8");
+      const markdown = decoder.decode(await Deno.readFile(postPath));
+      const markup = Marked.parse(markdown);
+
+      let github;
+      if (markup.meta.github) {
+        github = await fetchRepoMetadata(markup.meta.github);
+      }
+
+      return ctx.render({
+        post: { ...markup.meta, content: markup.content },
+        github,
+      });
+    } catch (error) {
+      // redirect to home page if post not found
+      return new Response(undefined, {
+        status: 302,
+        headers: { location: "/" },
+      });
+    }
+  },
+};
+
+export default function Blog({ data }: PageProps) {
```

Comment:
You could use the generic version of PageProps (PageProps<T>) to add type for your data object to keep later code clear of annotations, e.g. `PageProps<{ post: PostDetail, github: any }>` (i'm not sure what type github value is here)